src/pages/admin/ProductRequestPage/index.tsx

```diff
@@ -0,0 +1,116 @@
+import { useEffect, useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { NavLink } from 'react-router-dom';
+
+import { Box, CircularProgress } from '@mui/material';
+
+import { sortOptions, urls } from 'src/common/constants';
+import useErrorHandling from 'src/common/hooks/useErrorHandlingHook';
+import SearchInput from 'src/components/shared/SearchInput';
+import useToast from 'src/components/shared/toasts/components/ToastProvider/ToastProviderHooks';
+import { useGetAllProductsQuery } from 'src/redux/adminProduct/adminProductService';
+import { SortOrder } from 'src/redux/user/types';
+import theme from 'src/theme';
+
+import AdminSectionSubTitle from '../AdminSectionSubTitle';
+import AdminSectionTitle from '../AdminSectionTitle';
+import ProductsTable from '../ProductsTable';
+import SortButton from '../SortButton';
+
+import {
+  SectionWrapper,
+  StyledListItemButton,
+  StyledTypography,
+} from './styles';
+
+function ProductRequestPage() {
+  const { t } = useTranslation();
+  const { showToast } = useToast();
+  const methods = useForm();
+
+  const [page, setPage] = useState(1);
+  const handleChange = (_: React.ChangeEvent<unknown>, value: number) => {
+    setPage(value);
+  };
+
+  const [sortOrder, setSortOrder] = useState<SortOrder>(sortOptions.DESC);
+  const handleClick = (value: SortOrder) => {
+    setSortOrder(value);
+  };
+
+  const [search, setSearch] = useState('');
+  const handleSearchChange = (value: string) => {
+    setSearch(value);
+    setPage(1);
+  };
+
+  const { data, isLoading, error } = useGetAllProductsQuery({
+    list: 'inactive',
+    page,
+    sortOrder,
+    search,
+  });
+  const limit = 5;
+  const products = data?.products || [];
+  const count = data?.count || 1;
+  const pagesCount = Math.ceil(count / limit);
```

Comment:
I already see this logic, its mean that it could be combined and make reusable

Reply:
moved to custom hook